cmd/containerd/server/server.go

```diff
@@ -200,7 +201,19 @@ func New(ctx context.Context, config *srvconfig.Config) (*Server, error) {
 			tlsConfig.ClientCAs = caCertPool
 			tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
 		}
-
+		tcpServerOpts = append(tcpServerOpts, grpc.Creds(credentials.NewTLS(tlsConfig)))
+	} else if config.GRPC.TCPTLSCName != "" {
+		tlsConfig, CA, res, err :=
+			wintls.SetupTLSFromWindowsCertStore(ctx, config.GRPC.TCPTLSCName)
```

Comment:
If you already have per platform function here, just have a single call to setup `TCPTLSCname` and `setTLSResource` which only has an implementation on Windows. The `wintls` package may not even be necessary if it just used here, not sure where else it would be used at.

Reply:
I had started with the same thought initially, but then thought about tls handling for streamserver (https://github.com/containerd/containerd/blob/7d8765d440e4991a7d005db19e7873651ce3551d/internal/cri/config/streaming.go#L92) which serves container attach, exec etc. I have not implemented tls configuration using common name for stream server yet.
It is better to future proof by separating fetching of certificates given a common name and tls resource caching/cleanup for servers. Separate wintls can be useful for any other places in code which may want to act as client or server for tls traffic.